tests/functional/behave_features/common/utils/owner_file_submission.py

```diff
@@ -0,0 +1,351 @@
+        """
+        data = {
+            "head": pr_branch,
+            "base": base_branch,
+            "title": base_branch,
+            "body": pr_body,
+        }
+        logging.debug(f"Pull Request Body Text: {pr_body}")
+        logging.info(
+            f"Create PR from '{remote_repo}:{pr_branch}' to '{remote_repo}:{base_branch}'"
+        )
+        r = github.github_api("post", f"repos/{remote_repo}/pulls", gh_token, json=data)
+        j = json.loads(r.text)
+        if "number" not in j:
+            raise AssertionError(f"error sending pull request, response was: {r.text}")
+        return j["number"]
+
+    def check_workflow_conclusion(
+        self,
+        expected_result: str,  # the possible outcomes of a github workflow. e.g. 'success'
+    ):
+        """Checks the input workflow and reports back if expected_result doesn't match reality."""
+
+        # TODO(komish): When this testing is extended to partner/community files,
+        # setting workflow_name should be more dynamic. For now, this always looks
+        # for the Red Hat workflow because those are the only checks that exist.
+        workflow_name = settings.WORKFLOW_REDHAT_OWNERS_CHECK
+
+        return self._check_workflow_conclusion(
+            self.pull_request,
+            workflow_name,
+            expected_result,
+        )
+
+    def _check_workflow_conclusion(
+        self,
+        pr_number: str,  # '1'
+        workflow_name: str,
+        expected_result: str,  # 'success'
+        failure_type="error",
+    ):
+        """Checks the conclusion of workflow_name for expected_result via the GitHub API.
+
+        if failure_type is set to "error", this raises an exception.
+
+        Raises:
+            AssertionError if the conclusion does not match the expected_result.
+        """
+        try:
+            run_id = github.get_run_id(self.secrets, workflow_name, pr_number)
+            conclusion = github.get_run_result(self.secrets, run_id)
+            if conclusion == expected_result:
+                logging.info(
+                    f"PR{pr_number} Workflow run was '{expected_result}' which is expected"
+                )
+            else:
+                if failure_type == "warning":
+                    logging.warning(
+                        f"PR{pr_number if pr_number else self.secrets.pr_number} Workflow run was '{conclusion}' which is unexpected, run id: {run_id}"
+                    )
+                else:
+                    raise AssertionError(
+                        f"PR{pr_number if pr_number else self.secrets.pr_number} Workflow run was '{conclusion}' which is unexpected, run id: {run_id}"
+                    )
+
+            return run_id, conclusion
+        except Exception as e:
+            if failure_type == "error":
+                raise AssertionError(e)
+            else:
+                logging.warning(e)
+                return None, None
+
+    def set_chart_name(self, basename):
```

Comment:
Note that this is currently not called anywhere. Future planning ?

Reply:
Yes, but I'll pull it if it's not called.

Reply:
Scratch that, this is used in the behave implementation.